app/sidekiq/form0781_state_snapshot_job.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+# Log information about Form 0781 submissions to populate an admin facing Datadog dashboard
+class Form0781StateSnapshotJob
+  include Sidekiq::Job
+  sidekiq_options retry: false
+
+  STATSD_PREFIX = 'form0781.state.snapshot'
+
+  def perform
+    write_0781_snapshot
+  rescue => e
+    Rails.logger.error('Error logging 0781 state snapshot',
+                       class: self.class.name,
+                       message: e.try(:message))
+  end
+
+  def write_0781_snapshot
+    state_as_counts.each do |description, count|
+      StatsD.gauge("#{STATSD_PREFIX}.#{description}", count)
+    end
+  end
+
+  def state_as_counts
+    @state_as_counts ||= {}.tap do |abbreviation|
+      snapshot_state.each do |dp, ids|
+        abbreviation[:"#{dp}_count"] = ids.count
+      end
+    end
+  end
+
+  def snapshot_state
+    @snapshot_state ||= load_snapshot_state
+  end
+
+  def load_snapshot_state
+    {
+      # New 0781 form metrics
+      in_progress_new_0781_forms: new_0781_in_progress_forms,
```

Comment:
have the key in the object map the method name - prefix the key with the same prefix as the value